lib/bolt/transport/ssh.rb

```diff
@@ -122,19 +122,11 @@ def run_script(target, script, arguments, options = {})
       end
 
       def run_task(target, task, arguments, options = {})
-        if from_api?(task)
-          executable = task.file['filename']
-          file_content = Base64.decode64(task.file['file_content'])
-          input_method = task.metadata['input_method']
-          extra_files = []
-          write_executable = proc { |conn, dir| conn.write_executable_from_content(dir, file_content, executable) }
-        else
-          implementation = task.select_implementation(target, PROVIDED_FEATURES)
-          executable = implementation['path']
-          input_method = implementation['input_method']
-          extra_files = implementation['files']
-          write_executable = proc { |conn, dir| conn.write_remote_executable(dir, executable) }
-        end
+        implementation = task.select_implementation(target, PROVIDED_FEATURES)
+        executable = implementation['path']
+        input_method = implementation['input_method']
+        extra_files = implementation['files']
+        write_executable = proc { |conn, dir| conn.write_remote_executable(dir, executable) }
```

Comment:
The proc isn't really needed anymore, could move the call back to where it's used.